hpedockerplugin/file_manager.py

```diff
@@ -179,6 +129,47 @@ def create_share(self, share_name, **args):
         # Return success
         return json.dumps({"Err": ""})
 
+    def _get_existing_fpg(self, share_args):
+        cpg_name = share_args['cpg']
+        fpg_name = share_args['fpg']
+        try:
+            fpg_info = self._fp_etcd_client.get_fpg_metadata(
+                self._backend,
+                cpg_name, fpg_name
+            )
+        except exception.EtcdMetadataNotFound:
+            LOG.info("Specified FPG %s not found in ETCD. Checking "
+                     "if this is a legacy FPG..." % fpg_name)
+            # Assume it's a legacy FPG, try to get details
+            leg_fpg = self._hpeplugin_driver.get_fpg(fpg_name)
```

Comment:
Hi Imran, what if the fpg doesn't exist on 3par itself?

Reply:
Then the plugin assumes that user wants to create a new FPG by the specified name